platform/node/src/node_map.cpp

```diff
@@ -579,6 +582,82 @@ void NodeMap::RemoveLayer(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 
     nodeMap->map->removeLayer(*Nan::Utf8String(info[0]));
 }
+    
+void NodeMap::AddImage(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+    using namespace mbgl::style;
+    using namespace mbgl::style::conversion;
+    
+    auto nodeMap = Nan::ObjectWrap::Unwrap<NodeMap>(info.Holder());
+    if (!nodeMap->map) return Nan::ThrowError(releasedMessage());
+    
+    if (info.Length() != 3) {
+        return Nan::ThrowTypeError("Three arguments required");
+    }
+    
+    if (!info[0]->IsString()) {
+        return Nan::ThrowTypeError("First argument must be a string");
+    }
+
+    if (!info[1]->IsObject()) {
+        return Nan::ThrowTypeError("Second argument must be an object");
+    }
+    
+    if (!info[2]->IsObject()) {
+        return Nan::ThrowTypeError("Third argument must be an object");
+    }
+    
+    auto optionObject = Nan::To<v8::Object>(info[2]).ToLocalChecked();
+    
+    if (!Nan::Get(optionObject, Nan::New("height").ToLocalChecked()).ToLocalChecked()->IsUint32()) {
+        return Nan::ThrowTypeError("height parameter required");
+    }
+    
+    if (!Nan::Get(optionObject, Nan::New("width").ToLocalChecked()).ToLocalChecked()->IsUint32()) {
+        return Nan::ThrowTypeError("width parameter required");
+    }
+    
+    if (!Nan::Get(optionObject, Nan::New("pixelRatio").ToLocalChecked()).ToLocalChecked()->IsUint32()) {
+        return Nan::ThrowTypeError("pixelRatio parameter required");
+    }
+    
+    uint32_t imageHeight = Nan::Get(optionObject, Nan::New("height").ToLocalChecked()).ToLocalChecked()->Uint32Value();
+    uint32_t imageWidth = Nan::Get(optionObject, Nan::New("width").ToLocalChecked()).ToLocalChecked()->Uint32Value();
+    
+    if (imageWidth > 1024 || imageHeight > 1024) {
```

Comment:
FYI, it’s possible for the image-adding operation to have no effect (and log to stderr) if the size of all the images combined exceeds this 1024-by-1024 square.